pkg/vtpm/swtpm-vtpm/src/main.go

```diff
@@ -191,16 +206,23 @@ func handleLaunch(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err, http.StatusBadRequest)
 		return
 	}
-	// If we have SWTPM instance with the same id running, it means either the
-	// domain got rebooted or something went wrong on the dommain manager side!
-	// it the later case it should have sent a terminate request if VM crashed or
-	// there was any other VM related errors. Anyway, refuse to launch a new
+	// If we have a record of SWTPM instance with the requested id,
+	// check if it's still alive. if it is alive, refuse to launch a new
 	// instance with the same id as this might corrupt the state.
-	if _, ok := pids[id]; ok {
-		log.Warnf("SWTPM instance with id %s already running", id)
-		// technically request is satisfied
-		w.WriteHeader(http.StatusOK)
-		return
+	if p, ok := pids[id]; ok {
+		// don't trust the p, it might be a pid resused.
+		pidPath := fmt.Sprintf(swtpmPidPath, id)
+		if _, err := os.Stat(pidPath); err == nil {
+			pid, err := getSwtpmPid(pidPath, 0)
+			if err == nil || isAlive(pid) {
+				log.Warnf("SWTPM instance with id %s already running with pid %d", id, p)
```

Comment:
Does this warning make it to logread -F?

Reply:
I assumed it should, let me check and report.